To: vim_dev@googlegroups.com
Subject: Patch 7.3.260
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.260
Problem:    CursorHold triggers on an incomplete mapping. (Will Gray)
Solution:   Don't trigger CursorHold when there is typeahead.
Files:	    src/fileio.c


*** ../vim-7.3.259/src/fileio.c	2011-05-19 17:25:36.000000000 +0200
--- src/fileio.c	2011-07-20 18:22:16.000000000 +0200
***************
*** 9044,9050 ****
  {
      int		state;
  
!     if (!did_cursorhold && has_cursorhold() && !Recording
  #ifdef FEAT_INS_EXPAND
  	    && !ins_compl_active()
  #endif
--- 9044,9053 ----
  {
      int		state;
  
!     if (!did_cursorhold
! 	    && has_cursorhold()
! 	    && !Recording
! 	    && typebuf.tb_len == 0
  #ifdef FEAT_INS_EXPAND
  	    && !ins_compl_active()
  #endif
***************
*** 9935,9940 ****
--- 9938,9945 ----
  	    if ((c == ';' || c == '>') && match == FALSE)
  	    {
  		*pattern = NUL;	    /* Terminate the string */
+ 		/* TODO: match with 'filetype' of buffer that "fname" comes
+ 		 * from. */
  		match = mch_check_filetype(fname, type_start);
  		*pattern = c;	    /* Restore the terminator */
  		type_start = pattern + 1;
*** ../vim-7.3.259/src/version.c	2011-07-20 17:58:14.000000000 +0200
--- src/version.c	2011-07-20 18:25:15.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     260,
  /**/

-- 
LAUNCELOT: At last!   A call!  A cry of distress ...
           (he draws his sword, and turns to CONCORDE)
           Concorde!  Brave, Concorde ... you shall not have died in vain!
CONCORDE:  I'm not quite dead, sir ...
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
